packages/viewer/src/components/scene-tree/scene-tree.tsx

```diff
@@ -543,6 +544,25 @@ export class SceneTree {
     return this.controller?.filter(term, options);
   }
 
+  /**
+   * Performs an async request that will select the filtered items in the tree
+   * that match the given term.
+   *
+   * @param term The filter term.
+   * @returns A promise that resolves with the selected rows.
+   */
+  @Method()
+  public async selectFilteredItems(term: string): Promise<Row[] | void> {
+    if (this.viewer != null) {
+      await selectFilterResults(this.viewer, term, this.metadataKeys, {
+        append: false,
+      });
+      return this.rows.filter((row) => row?.node.filterHit);
```

Comment:
I don't think we should return the filtered rows from this operation. The client doesn't have the full list of rows, so this result is very likely to be incomplete. If for some reason the caller needs this list, we'll have to find another way to return it to them.